src/types/query.ts

```diff
@@ -4,6 +4,8 @@ import { Param } from '@/types/paramTypes'
 import { Identity } from '@/types/utilities'
 import { isRecord } from '@/utilities/guards'
 
+export type QueryRecord = ConstructorParameters<typeof URLSearchParams>[0]
```

Comment:
I tested this out and I think we need to just copy the type from the constructor rather than trying to reference it. This type here gets widened by typescript and ends up accepting some invalid input. For example it thinks this value is fine when it should error (and does error for URLSearchParams)
<img width="750" alt="image" src="https://github.com/user-attachments/assets/c951a760-e9fc-4d48-8e90-392461ab63e5">

Lets do something like this
```ts
type QuerySource =
  | URLSearchParams
  | string
  | Record<string, string | readonly string[]>
  | Iterable<[string, string]>
  | readonly [string, string][]
```

Suggesting `QuerySource` because `QueryRecord` implies its only a record.